src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotDeleteSamGroupStep.java

```diff
@@ -27,20 +27,19 @@ public class DeleteSnapshotDeleteSamGroupStep extends DefaultUndoStep {
    * @param iamService
    * @param snapshotId
    */
-  public DeleteSnapshotDeleteSamGroupStep(IamService iamService, UUID snapshotId) {
+  public DeleteSnapshotDeleteSamGroupStep(
+      IamService iamService, SnapshotRequestDao snapshotRequestDao, UUID snapshotId) {
     this.iamService = iamService;
+    this.snapshotRequestDao = snapshotRequestDao;
     this.snapshotId = snapshotId;
   }
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    List<String> authDomains =
-        context
-            .getWorkingMap()
-            .get(SnapshotWorkingMapKeys.SNAPSHOT_AUTH_DOMAIN_GROUPS, new TypeReference<>() {});
-    // Only delete the Sam group if it matches the expected naming pattern
-    var expectedName = IamService.constructSamGroupName(snapshotId.toString());
-    if (Objects.nonNull(authDomains) && authDomains.contains(expectedName)) {
+    // Only delete the Sam group if it matches the expected name
+    SnapshotAccessRequestModel request = snapshotRequestDao.getByCreatedSnapshotId(snapshotId);
+    if (Objects.nonNull(request)) {
```

Comment:
In general, I think methods like `Objects.nonNull` and `Objects.isNull` are intended for use for passing to methods like `Stream.filter()`. In cases like this I think it's clearer to compare against `null` directly
```suggestion
    if (request != null) {
```